From 05edbc5606ae2f7fbe9cfb7441dd163afea9e8a9 Mon Sep 17 00:00:00 2001
From: Wade Cline <wadecline@hotmail.com>
Date: Fri, 22 Dec 2017 20:38:29 -0800
Subject: [PATCH 1/7] Fix OpenSSL 1.0.x wrapper.

Make OPENSSL_init_ssl() a wrapper to a function so it can return a value
and be called within another expression.
---
 src/modules/extra/m_ssl_openssl.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/modules/extra/m_ssl_openssl.cpp b/src/modules/extra/m_ssl_openssl.cpp
index 57af1316d..6a0cd4a29 100644
--- a/src/modules/extra/m_ssl_openssl.cpp
+++ b/src/modules/extra/m_ssl_openssl.cpp
@@ -93,9 +93,13 @@
 # define OpenSSL_version SSLeay_version
 # define X509_getm_notAfter X509_get_notAfter
 # define X509_getm_notBefore X509_get_notBefore
-# define OPENSSL_init_ssl(OPTIONS, SETTINGS) \
-	SSL_library_init(); \
+# define OPENSSL_init_ssl(OPTIONS, SETTINGS) OPENSSL_init_compat()
+static int OPENSSL_init_compat() {
+	SSL_library_init();
 	SSL_load_error_strings();
+	ERR_load_X509_strings();
+	return 1;
+}
 
 // These macros have been renamed in OpenSSL 1.1.
 # define OPENSSL_VERSION SSLEAY_VERSION
-- 
2.32.0

